File: src/checkers.frontend/src/components/game/piece.js
import React from "react";

import "../../styles/piece.css";

export default class Piece extends React.Component{
	render(){
		return (
			<div className={`piece 
			${this.props.source.colour === 0 ? "piece-white" : "piece-black"} 
			${this.props.source.isKing ? "piece-king" : ""} 
			${this.props.selected ? "piece-selected" : ""}`}>
				<div className="piece-inside">
					{this.props.source.isKing &&
					<img src="/crown.svg" alt="king"/>}
				</div>
			</div>
		);
	}
}

File: src/Draughts.Api/Draughts/Board/Piece.cs
namespace Draughts.Api.Draughts
{
    public class Piece
    {
        public PieceColour Colour { get; set; }
        public bool IsKing { get; set; }

        public Piece(PieceColour colour, bool isKing = false)
        {
            Colour = colour;
            IsKing = isKing;
        }

        public Piece Clone()
        {
            return new(Colour, IsKing);
        }
        
        public static Piece White()
            => new(PieceColour.White);
        
        public static Piece Black()
            => new(PieceColour.Black);
        
        public static Piece WhiteKing()
            => new(PieceColour.White, true);
        
        public static Piece BlackKing()
            => new(PieceColour.Black, true);
    }
}
